lua/nvim_lsp/metals.lua

```diff
@@ -0,0 +1,79 @@
+local skeleton = require 'nvim_lsp/skeleton'
+local util = require 'nvim_lsp/util'
+local lsp = vim.lsp
+local server_name = "metals"
+local bin_name = "metals"
+
+local function make_installer()
+  local install_dir = util.path.join{util.base_install_dir, server_name}
+  local metals_bin = util.path.join{install_dir, bin_name}
+  local X = {}
+  function X.install()
+    local install_info = X.info()
+    if install_info.is_installed then
+      print(name, "is already installed")
+      return
+    end
+    if not (util.has_bins("curl")) then
+      error('Need "curl" to install this.')
+      return
+    end
+    if not (util.has_bins("java")) then
+      error('Need "JDK" to install this.')
+      return
+    end
+    local coursier_bin = install_dir .. "/coursier"
+    local download_cmd = string.format("curl -fLo %s --create-dirs https://git.io/coursier", coursier_bin)
+    local chmod_cmd = string.format("chmod +x %s", coursier_bin)
+    local install_cmd = string.format("%s bootstrap --java-opt -Xss4m --java-opt -Xms100m --java-opt -Dmetals.client=coc.nvim org.scalameta:metals_2.12:0.7.6 -r bintray:scalacenter/releases -r sonatype:snapshots -o %s -f", coursier_bin, metals_bin)
```

Comment:
You may want to take a quick look at to make sure you want `Dmetals.client=coc.nvim` since that's determining some of the functionality and the way we return stuff to the client. `vim-lsc` is also an option for now which might be closer depending on how you are handing messages coming from the server. If need be, we can totally also create a new case specifically for `nvim.lsp` or something if you need other settings. If it's just additional settings, then you can just use an existing and pass them in here as well.

Here is a reference to where these are read on the server side.
https://github.com/scalameta/metals/blob/master/metals/src/main/scala/scala/meta/internal/metals/MetalsServerConfig.scala#L111-L132

Reply:
Never mind about my question with configuration options, I'm not seeing the info about settings in `default_config`

Reply:
Hmm, I do not want to add `-Dmetals.client` because the builtin LSP support is still under development, but someone will add this option in the future when builtin LSP support becomes stable.